interface/src/Application.cpp

```diff
@@ -5814,10 +5816,10 @@ void Application::resumeAfterLoginDialogActionTaken() {
 
         // if the --scripts command-line argument was used.
         if (_overrideDefaultScriptsLocation && _defaultScriptsLocation.exists()) {
-            scriptEngines->loadDefaultScripts();
+            scriptEngines->loadScripts();
```

Comment:
Wait, why did you swap this around?

My comment was that the fact that different functions get called in different branches here is weird. You'd expect that an option that changes a path just changes a path, not results in different behaviors being implemented.

That is, what I would have expected the code to look like here is something along these lines:

```
if (_overrideDefaultScriptsLocation && _defaultScriptsLocation.exists()) {
    scriptEngines->setPath(_defaultScriptsLocation);
}
scriptEngines->loadScripts();
```

Looking at the code:

`scriptEngines->loadDefaultScripts();` -- this works out to `loadScript(DEFAULT_SCRIPTS_LOCATION);`, where DEFAULT_SCRIPTS_LOCATION is a constant.

`scriptEngines->loadScripts()` loads a list of scripts from what's listed in the config file.

I think it was better the way it was before: the default path is to load whatever the user had decided.

Reply:
At this point I think half this stuff is probably useless to us, and a whole bunch of it is confusingly named.

I think the important part here is not to break anything important to us, and then do a bit of further cleanup in a future PR where we figure out better names for some options and maybe outright remove some of them.

Reply:
Ok. I agree. There are probably other parameters that would be useful and I'm sure someone will want to change some of the descriptions.

I will swap the functions back.

Reply:
The code you suggested seems much better to me but I don't want to break anything that might get confused with other changes in this pull request. Reconciling these script parameters could be another.

Reply:
No, I was just explaining why I made the original comment. I wasn't saying "swap this around", but "this looks kinda weird".

On a second thought, I think I get the reason why it was done this way now. If you override the path, then we only load the default, hardcoded scripts because that's all we can expect to be present at whatever path the user passed in.

If we're not overriding anything, then we have the normal function where the interface remembers what scripts the user wanted in its config file.

So I think the right way is just leaving it how it was before

Reply:
Ok. I think we should open an issue to reconcile the script parameters in a different PR. It would be good to reconsider them all, including their names, descriptions, and desired functionality.